AI/Nullkiller/AIUtility.cpp

```diff
@@ -393,11 +393,12 @@ bool shouldVisit(const Nullkiller * ai, const CGHeroInstance * h, const CGObject
 		}
 		return false;
 	}
-	case Obj::MONOLITH_ONE_WAY_ENTRANCE:
 	case Obj::MONOLITH_ONE_WAY_EXIT:
+		return false;
+	case Obj::MONOLITH_ONE_WAY_ENTRANCE:
 	case Obj::MONOLITH_TWO_WAY:
 	case Obj::WHIRLPOOL:
-		return false;
```

Comment:
This should be left as is. The mechanics of entering one-way monolith is a bit different and should relay on pathfinder instead. You can check my PR regarding whirlpools #4313 .